compiler/testData/codegen/bytecodeText/deadCodeElimination/boxing.kt

```diff
@@ -2,16 +2,16 @@ class A
 
 fun foo(x: Any?) {}
 
-fun box() {
+fun box(u: Int) {
```

Comment:
Why you touch this test?

Reply:
Similar to changes in concat.kt and primitiveToString.kt, this prevents the relevant part of the test being optimized away. Without this change, it would be:

expected:
<0 IFNULL
[1 IFNONNULL
1] throwNpe
0 ATHROW
> but was:
<0 IFNULL
[0 IFNONNULL
0] throwNpe
0 ATHROW

Namely, the check and potential throw for z!! would be optimized away.

Reply:
@ting-yuan I would prefer to add new tests with suggested changes (for this one and below). And for kt2269.kt

Reply:
@ting-yuan change in concat.kt is still totatly unclear for me, maybe this change is redundant after pr2132.
For primitiveToString.kt it would be best to add new test with suggested change

Reply:
@max-kammerer The test in concat.kt has a flew IMHO. Because the constants get folded into one, there is only one concatenation after constant folding (in both backends): "12345.06.07" + A(). Because the test looks for exactly 1 StringBuilder, it does not test whether the unnecessary StringBuilders are eliminated or not.

Reply:
@max-kammerer I touched kt2269.kt because :compiler:test fails and complains that kt2269.kt is a passing test disabled. So I have either to:
1. remove // IGNORE_BACKEND: JVM_IR so that it can pass falsely. I'll add kt2269-jvm-ir.kt
2. change the test as in PR to make it test what it meant to test.

Should I just go for 1, or other recommendations?

Reply:
@max-kammerer I've forked those tests with proposed changes and disabled the original tests for JVM_IR, if they fail. Basically testABC.kt is forked into testABCNotOptimizable.kt.